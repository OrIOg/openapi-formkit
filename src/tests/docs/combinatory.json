{
    "openapi": "3.0.0",
    "info": {
      "title": "Simple API overview",
      "version": "1.0"
    },
    "paths": {
        "/tests/allOf-unique": {
            "post": {
                "operationId": "combinatory",
                "requestBody": {
                    "content": {
                        "multipart/form-data": {
                            "schema": {
                                "$ref": "#/components/schemas/MatchingParameters"
                            }
                        }
                    },
                    "required": true
                }
            }
        },
        "/tests/allOf-multiple": {
            "post": {
                "operationId": "combinatory",
                "requestBody": {
                    "content": {
                        "multipart/form-data": {
                            "schema": {
                                "$ref": "#/components/schemas/MultipleAllOf"
                            }
                        }
                    },
                    "required": true
                }
            }
        }
    },
    
    "components": {
        "schemas": {
            "MatchingParameters": {
                "title": "ImageMatchingParameters",
                "type": "object",
                "required": ["matching_parameters"],
                "properties": {
                    "match_colors_with": {
                        "allOf": [{
                            "$ref": "#/components/schemas/Match"
                        }],
                        "default": 2
                    }
                }
            },
            "Match": {
                "title": "Match",
                "enum": [0, 1, 2, 3],
                "type": "integer",
                "description": "An enumeration."
            },


            "MultipleAllOf": {
                "title": "ImageMatchingParameters",
                "type": "object",
                "required": ["matching_parameters"],
                "properties": {
                    "match_colors_with": {
                        "allOf": [{
                            "$ref": "#/components/schemas/Cat"
                        },{
                            "$ref": "#/components/schemas/Dog"
                        }]
                    }
                }
            },
            "Cat": {
                "type": "object",
                "required": ["name", "age", "evil"],
                "properties": {
                    "name": {
                        "type": "string",
                        "exclusiveMinimum": 1
                    },
                    "age": {
                        "type": "integer",
                        "minimum": 0,
                        "format": "int32"
                    },
                    "evil": {
                        "type": "boolean"
                    }
                }
            },
            "Dog": {
                "type": "object",
                "required": ["name", "age", "goodboi"],
                "properties": {
                    "name": {
                        "type": "string",
                        "exclusiveMinimum": 1
                    },
                    "age": {
                        "type": "integer",
                        "minimum": 0,
                        "format": "int32"
                    },
                    "goodboi": {
                        "type": "boolean"
                    }
                }
            }
        }
    }
}