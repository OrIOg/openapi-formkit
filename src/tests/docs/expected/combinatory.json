{
  "/tests/allOf-unique": {
    "post":  {
      "$cmp": "FormKit",
      "children":  [
         {
          "$cmp": "FormKit",
          "props":  {
            "help": "An enumeration.",
            "label": "Match",
            "name": "match_colors_with",
            "options":  [
              0,
              1,
              2,
              3
            ],
            "type": "select",
            "value": 2
          }
        }
      ],
      "props":  {
        "method": "post",
        "type": "form"
      }
    }
  },
  "/tests/allOf-multiple": {
    "post": {
      "$cmp": "FormKit",
      "children": [
       {
         "$formkit": "group",
           "children": [
             {
               "$cmp": "FormKit",
               "props": {
                 "label": "name",
                 "name": "name",
                 "type": "text",
                 "validation": [
                   [
                     "required"
                   ]
                 ]
               }
             },
             {
               "$cmp": "FormKit",
               "props": {
                 "label": "age",
                 "min": 0,
                 "name": "age",
                 "step": 1,
                 "type": "number",
                 "validation": [
                   [
                     "required"
                   ]
                 ]
               }
             },
             {
               "$cmp": "FormKit",
               "props": {
                 "label": "evil",
                 "name": "evil",
                 "type": "checkbox"
               }
             },
             {
               "$cmp": "FormKit",
               "props": {
                 "label": "goodboi",
                 "name": "goodboi",
                 "type": "checkbox",
                 "validation": [
                   [
                     "required"
                   ]
                 ]
               }
             }
           ],
           "name": "match_colors_with",
           "props": {}
         }
       ],
       "props": {
         "method": "post",
         "type": "form"
       }
     }
   }
}